### PR TITLE
Enable to save key_pair in Authentication

### PR DESCRIPTION
**Pull request by hsong-rh**
Enable to save key_pair information in authentication table. Smart state analysis will use it to transfer files to the agent.

### CODE REVIEW

**Comment by hsong-rh** [comment]
@roliveri @jerryk55 @bronaghs Please review.

File: app/models/manageiq/providers/amazon/cloud_manager/auth_key_pair.rb [hunk 1]
```diff
@@ -1,2 +1,37 @@
 class ManageIQ::Providers::Amazon::CloudManager::AuthKeyPair < ManageIQ::Providers::CloudManager::AuthKeyPair
+  def self.raw_create_key_pair(ext_management_system, create_options)
+    ec2 = ext_management_system.connect
+    kp = ec2.create_key_pair(create_options)
+    ost = OpenStruct.new
```
Comment (roliveri) [inline app/models/manageiq/providers/amazon/cloud_manager/auth_key_pair.rb]: Can we use a hash or Struct here?

Comment (hsong-rh) [inline app/models/manageiq/providers/amazon/cloud_manager/auth_key_pair.rb]: Changed it to use Struct.

**Comment by bronaghs** [comment]
Hi @hsong-rh - can you set the labels, thanks.

**Pull request merged**